### PR TITLE
fix: use a linked list for process clearing

### PR DESCRIPTION
**Pull request by xavierchanth**
**- What I did**

- Switched the process id queue for srv from a dynamically allocated pointer array to a linked-list so memory is managed more effectively
  - This queue is used to clean up defunct processes - nearly immediately, unless the main thread is the in the middle of handling another request (which is usually a few seconds of delay at most)

**- How I did it**

**- How to verify it**

**- Description for the changelog**
fix: use a linked list for process clearing

### CODE REVIEW

File: packages/c/sshnpd/include/sshnpd/sshnpd.h [hunk 1]
```diff
@@ -32,9 +32,9 @@ enum notification_key {
   NK_NPT_REQUEST,
 };
 
-struct sshnpd_process_queue {
-  size_t len;
-  pid_t *processes;
+struct sshnpd_process_node {
+  pid_t process;
+  struct sshnpd_process_node *next;
```
Comment (xavierchanth) [inline packages/c/sshnpd/include/sshnpd/sshnpd.h:37]: Linked list node representation

File: packages/c/sshnpd/src/handle_ssh_request.c [hunk 2]
```diff
@@ -485,17 +486,27 @@ void handle_ssh_request(atclient *atclient, pthread_mutex_t *atclient_lock, sshn
...
+    struct sshnpd_process_node *pid_node = malloc(sizeof(struct sshnpd_process_node));
+    if (pid_node == NULL) {
+      atlogger_log(LOGGER_TAG, ATLOGGER_LOGGING_LEVEL_ERROR,
+                   "Unable to allocate memory to keep track of the srv process id");
       goto cancel;
     }
-    pids->processes[pids->len++] = pid;
+
+    pid_node->process = pid;
+    pid_node->next = NULL;
```
Comment (XavierChanth) [inline packages/c/sshnpd/src/handle_ssh_request.c:497]: Allocate new linked list node

**Comment by xavierchanth** [comment]
cc: @realvarx - Changing to the new parse_control_message approach has created (what I think is) a race condition in non `--multi` srv:

`[ERRO] 2024-07-30 22:06:01.229578 | srv - run | Failed to parse message type`

Previously the race condition was only possible with `--multi` enabled, but now it is occurring with `ssh_request` since the race condition happens without `--multi`.

**Comment by realvarx** [comment]
> cc: @realvarx - Changing to the new parse_control_message approach has created (what I think is) a race condition in non `--multi` srv:
> 
> `[ERRO] 2024-07-30 22:06:01.229578 | srv - run | Failed to parse message type`
> 
> Previously the race condition was only possible with `--multi` enabled, but now it is occurring with `ssh_request` since the race condition happens without `--multi`.

I didn't see that during my manual testing or cicd. Can you indicate the steps to reproduce? Or does it happen sometimes randomly?

**Comment by xavierchanth** [comment]
> cc: @realvarx - Changing to the new parse_control_message approach has created (what I think is) a race condition in non `--multi` srv:
> 
> 
> 
> `[ERRO] 2024-07-30 22:06:01.229578 | srv - run | Failed to parse message type`
> 
> 
> 
> Previously the race condition was only possible with `--multi` enabled, but now it is occurring with `ssh_request` since the race condition happens without `--multi`.

Following this, I will release v5.6.0 and c0.1.9, replace d:5.5.0 with d:5.6.0 in the e2e tests and raise the minimum C daemon compatibility to d:5.6.0.

**Comment by xavierchanth** [comment]
> > cc: @realvarx - Changing to the new parse_control_message approach has created (what I think is) a race condition in non `--multi` srv:
> > `[ERRO] 2024-07-30 22:06:01.229578 | srv - run | Failed to parse message type`
> > Previously the race condition was only possible with `--multi` enabled, but now it is occurring with `ssh_request` since the race condition happens without `--multi`.
> 
> I didn't see that during my manual testing or cicd. Can you indicate the steps to reproduce? Or does it happen sometimes randomly?

Happening randomly

**Review by JeremyTubongbanua — APPROVED** [review]

**Comment by JeremyTubongbanua** [comment]
LGTM

**Comment by JeremyTubongbanua** [comment]
Did my own manual test with M2 Air and Linux x64 VPS

```
v5.5.0 ./sshnp -f @soccer0 -t @soccer99 -r @rv_eu -d jeremyvps  
2024-07-31 11:41:28.918720 : Sending daemon feature check request
2024-07-31 11:41:28.918805 : Resolving remote username for user session
2024-07-31 11:41:29.844605 : Resolving remote username for tunnel session
2024-07-31 11:41:29.845382 : Fetching host and port from srvd
2024-07-31 11:41:32.775746 : Received host and port from srvd
2024-07-31 11:41:32.776020 : Waiting for daemon feature check response
2024-07-31 11:41:32.776053 : Received daemon feature check response
2024-07-31 11:41:32.778316 : Required daemon features are supported
2024-07-31 11:41:32.942082 : Sending session request to the device daemon
2024-07-31 11:41:33.016664 : Waiting for response from the device daemon
2024-07-31 11:41:33.624448 : Received response from the device daemon
2024-07-31 11:41:33.632024 : Creating connection to socket rendezvous
2024-07-31 11:41:33.780441 : Starting user session
The authenticity of host '[localhost]:49684 ([127.0.0.1]:49684)' can't be established.
ED25519 key fingerprint is SHA256:5ZBwC7F9UmJBDpiOo4vkIzyiBMI0sgfoXK++7xUuRUI.
This host key is known by the following other names/addresses:
    ~/.ssh/known_hosts:3: jeremymark.ca
    ~/.ssh/known_hosts:12: [localhost]:56249
    ~/.ssh/known_hosts:13: [localhost]:56272
    ~/.ssh/known_hosts:14: [localhost]:56374
    ~/.ssh/known_hosts:18: [localhost]:54353
    ~/.ssh/known_hosts:19: [localhost]:54572
    ~/.ssh/known_hosts:27: [localhost]:55657
    ~/.ssh/known_hosts:31: [localhost]:221
    (7 additional names omitted)
Are you sure you want to continue connecting (yes/no/[fingerprint])? ^C%
```

```
[DEBG] 2024-07-31 15:41:29.604226 | sshnpd | Waiting for next monitor thread message
[DEBG] 2024-07-31 15:41:30.214765 | atclient_monitor | 	RECV: "{"id":"a3a348e3-a45c-487e-aeef-7a858670d964","from":"@soccer0","to":"@soccer99","key":"@soccer99:ping.jeremyvps.sshnp@soccer0","value":"LxOUMaCIAzvu6j+Y/BhcoQ==","operation":"update","epochMillis":1722440490140,"messageType":"MessageType.key","isEncrypted":true,"metadata":{"encKeyName":null,"encAlgo":null,"ivNonce":"ZxKmERmPkSEekjQQqeP/OA==","skeEncKeyName":null,"skeEncAlgo":null}}"
[DEBG] 2024-07-31 15:41:30.214844 | sshnpd | Locked the atclient
[DEBG] 2024-07-31 15:41:30.214916 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:30.246882 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:30.246969 | connection | 	SENT: "lookup:shared_key@soccer0"
[DEBG] 2024-07-31 15:41:30.278389 | connection | 	RECV: "data:dGqHeGlk00BIp0Fj9LG3MXjFXRPpi7HG/wmBRvupQA6G2WqTMENDSjR9afp10Io3MQnyhnEgB0XnhlXMnhj6vCf5BvoynOCxfU8U/rnEtRKD+H3rdxWgEr+343JWqckTMTOacsqiOqfgZ7HytEFWeIE5lhKb2EG/xvnJjBqZDq8mZP83TGI3T5UtGSxY2CI29Zjf330a762ruskf/QeeLaAVFCtDDRKaNAGz3MnDnkQw347gnsPZjvopw0j9I3PyqG9lK5kehm/+/Ha7WH/AmgfPOTDWcWR5BoCNJ+bSTja1/IzzfqwXSvgj54JEXbrAdIggRZsJYEiSQtcXxlBRHw=="
[DEBG] 2024-07-31 15:41:30.292429 | sshnpd | Released the atclient lock
[DEBG] 2024-07-31 15:41:30.292467 | sshnpd | Received message of type: 1
[DEBG] 2024-07-31 15:41:30.292472 | sshnpd | Notification value received: ping
[DEBG] 2024-07-31 15:41:30.292479 | sshnpd | Executing handle_ping
[DEBG] 2024-07-31 15:41:30.292557 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:30.322409 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:30.322501 | connection | 	SENT: "llookup:shared_key.soccer0@soccer99"
[DEBG] 2024-07-31 15:41:30.352679 | connection | 	RECV: "data:GRluAnPbZML0MVUCoq/MWwv0ZVZyRNigrKoeMyn2pZxBujbfPk+KAoGmOmwvEJGsYuplAKZLLreV3QcQxHxUaHQ/weWwXMrgaogJkUlSt99fUAl/vZmK0hG31tnyJ4suEQTJHAUlmc09fy1mLlP/oBZuRLijL9aH0XirGsd7wAoFFuNNaKwhN0Lx8mVjrQNqq4uY39pJuMWAPebLG6uNQk6PZX8GQ9tu15YGerJqRJUsVrXVkPq3jNmRMaMjZWbluz/bzD7b7BK207WqiMv938awGD0qfixPjVg8gY2kY+8uMMXYlVe8eZz4C0LC/egFeEXJkidY2FeNCeJ8oIMtgQ=="
[DEBG] 2024-07-31 15:41:30.368701 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:30.398603 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:30.398697 | connection | 	SENT: "notify:update:messageType:key:priority:low:strategy:all:ttln:86400000:ttl:10000:isEncrypted:true:ivNonce:g3KOSaKHsgF9CCbjLgqnAw==:@soccer0:heartbeat.jeremyvps.sshnp@soccer99:P1bbzutJLaYjmqrWazLjSz/F95hmBgJQj14x0JNGjE3riTP0F6d4XL9yGgUE+8Dj3cbLJeiIyczKxbE7k3TyFUJ2POTvgZyy1NBZskov3C6c91UdednIMRYkTk/AhWEZJBfgYbek1E82Cqi8Ik0Yeq7CqCvEwqTno+9yIqCV1wqVw+aZDsYRfqzqFzF2/zjGisbejrTSwBVzESWZMDT54DfcJ6sKvpsGnEhodZJC2IDRASsUECRItDLujeEf82z+v3iXy3DI2PnSA+AcCpc527sH5ZeBUxQ+FA6W61NgH6f+LXvPUK3GHEQAujOeG6vk"
[DEBG] 2024-07-31 15:41:30.429052 | connection | 	RECV: "data:78799673-c6ff-4c66-846d-75958492c46f"
[DEBG] 2024-07-31 15:41:30.429096 | PING RESPONSE | Released the atclient lock
[DEBG] 2024-07-31 15:41:30.429109 | sshnpd | Waiting for next monitor thread message
[DEBG] 2024-07-31 15:41:33.113273 | atclient_monitor | 	RECV: "{"id":"52c1904a-bf83-4557-b670-55e60eaa25e8","from":"@soccer0","to":"@soccer99","key":"@soccer99:ssh_request.jeremyvps.sshnp@soccer0","value":"tSYxuiiNgdwUqDRGsrSIoqfeQrBsGgYNUvnqiijknHkL4cKakrW8umK5mBjeSz6YP5moKcT8eRDAyp+TXX23gEGyMyG2h8Rk1V0np4CiUpZTAUTsupgbRWqgs7j/JJqLGf+T8pholZKQmCLvxbCB49KAvbBtbn9ouJ3fDV3F25MGKcuxCDya/NJB5xmo7KKedMVFXMJ3AX6H/HDM1SMtiysgxCOhcfgAYN8USJGHL3jaz9zkH7W6QkriOjW2ojts+qdobZLAu2AvYwgaqKzAYGPKQvHDTvdZestgo/oEZA6VeYyAF8LdH87vY2tHp8VL+LQPYee78GgcUZV0SQvKxbnrFozNniR2TsLWrD6xfS6GHbGkRXtgvf+QjMdLSam9ZZ/L2p4cPzssduA96msWE/ejg3vYdOUImuOHEMs3a+FMvz4Ry0vkTPsOzKm5MCRQJyqKL7DYk0mcDsIlUwpfT8gmLqtj+alJWh6Rnv6PN1IHueyXZwKVU0jCI6giRQ61RV3OzhUTiUniRczxJTkiJd7dO3f+NVaGZz+bPVWC9esWtetgfcmoaG1m45fxtkS44Ua01QapQstdnVEeqA1zgBLHpb2pTM0O6v5hBDFus96et8zxgkp3ixuw97JgaeB3YIoeHrlmcnFRghvlpfjWjd0Td6dxPjTfay2Bk2/QdCyDgMlAETeJqPrgRKOpTyMjmayaX32emWXuZxCH8HC5z7vAZTHal1a0iQXTnrfiRnXQpz6k18FL9446JPNzAZOXW6QPsDD9JoCBvP4dQjcIWPAdl1q6G4PO713G6VF+2avqHC3LnFM5D4cSvwyn0MN9HyUKNtfNjeV+DMo7GkFPrxbT2srY1IDXzGlYnxBArmhqKcOSglcKiH8hY/2n5blnwvAMy3JICeLlysIyEDyCewTsxYJd+ZkTHZl5dqbafu8N2HHXKt7miZuS1RIX1JO11p325Rxg6nzy/C8iWYDkXw3PxmzfBLmaEjAP5LHBxW6Weo50Uw7Y2awLa9fZNkm3mf9c0IvZkgzhkWRK+P0tGM9W+ZkZA4J68sSwK1CofykT4NAKa0zIsKhWuOYqEM1Wybi/3koaniytZv+2q2VqHN2kwIXDESiOr9AEo8+iUut6wuzwIpvMMqs/7qp0eatGdE+6n5dtN/E3DKLl90tKyY87nfQ/j25neNyQPw4fYcshORhTBLy1HpFDLuStXThH8z/pmqAKEgTfWHkgxLSW7ybuCC+1T76eoIGkShioRlpeKbf//DBbNLi9w6kQ9u7gTQINijn1YuUSzcHZO/LrXkbVmTsLSmNw03G6+gFL0UsbYVXirtjtg8kQScfukKaNxIi3VQqKV1mpfasJF2Kb8cGAJMn2lhcOaZXs2HVX/fdoP5nDt7AmfRMA7bTgCy+25DFkWhQPDshD8yhWRCmHt94Bi2y/q1+UAF7DE1kEpHdi2K1hJ8iTvJCwJOGr7/mU","operation":"update","epochMillis":1722440493042,"messageType":"MessageType.key","isEncrypted":true,"metadata":{"encKeyName":null,"encAlgo":null,"ivNonce":"b4EEoS789eOad78K1ps34A==","skeEncKeyName":null,"skeEncAlgo":null}}"
[DEBG] 2024-07-31 15:41:33.113394 | sshnpd | Locked the atclient
[DEBG] 2024-07-31 15:41:33.113458 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:33.143587 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:33.143648 | connection | 	SENT: "lookup:shared_key@soccer0"
[DEBG] 2024-07-31 15:41:33.174297 | connection | 	RECV: "data:dGqHeGlk00BIp0Fj9LG3MXjFXRPpi7HG/wmBRvupQA6G2WqTMENDSjR9afp10Io3MQnyhnEgB0XnhlXMnhj6vCf5BvoynOCxfU8U/rnEtRKD+H3rdxWgEr+343JWqckTMTOacsqiOqfgZ7HytEFWeIE5lhKb2EG/xvnJjBqZDq8mZP83TGI3T5UtGSxY2CI29Zjf330a762ruskf/QeeLaAVFCtDDRKaNAGz3MnDnkQw347gnsPZjvopw0j9I3PyqG9lK5kehm/+/Ha7WH/AmgfPOTDWcWR5BoCNJ+bSTja1/IzzfqwXSvgj54JEXbrAdIggRZsJYEiSQtcXxlBRHw=="
[DEBG] 2024-07-31 15:41:33.189493 | sshnpd | Released the atclient lock
[DEBG] 2024-07-31 15:41:33.189542 | sshnpd | Received message of type: 1
[DEBG] 2024-07-31 15:41:33.189547 | sshnpd | Notification value received: {"payload":{"direct":true,"sessionId":"28dee05d-ce71-4d24-af38-638233a495e4","host":"85.239.63.180","port":39923,"authenticateToRvd":true,"clientNonce":"2024-07-31T11:41:28.916345","rvdNonce":"2024-07-31T15:34:02.419624","encryptRvdTraffic":true,"clientEphemeralPK":"MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnOr+b1hqQrJk0RpvxXmxtHlFBclw6VJH1cS4KK+NhzlrFWUzU4irGcEW5y/XS1N5j38cZ7hsdnOo7EAy375+nEU0sbqUSVoYjpCBkvl44tDg188jbeBKCSLye4kYdHDi8pfVlDR//yiODkJKxIMJ3Xaq/Wbkd98IZlkwHupF8Ii/8g9/8WYjLCZqEGlrqGFwr2t89b8n1xLHuIHhrbabp+XIXoT3J+axquCjxxzGZOgpPX8AxQhlA49d3i93E2y0PGHHxu8o2sRVEHRZsDlw82h18gRIeIaFxTT+6SDlkt21SeZBbpoR+QaaQJ4FqbEZZLPgHjB8khvEI8dc9p+1mQIDAQAB","clientEphemeralPKType":"rsa2048"},"signature":"MQC66pDXZrxYLMLjEKt/S/QIKoS7+7sUGzdSCZ+NJOzOm953u9Wx828vbsCvXOFIPxob9qwdSSCakTD/nUPy/+BZtpGqPBxbNK0vTDBO+KoVENPpy3XhyKeEY9BZOrAn9RsVtio+U9YT/TIW08j4t92+rFaOk/PwCPW1rslQzQT8a+wLthDDf965hZpsBWIeto06ksvr4ChiGwvHQQOdbU9Ha0DTQzSSoMoKU9UBAZBRVC/5VY9VRr/6d2WLn0sCOHpUI75kI83WZjQVOEb9NTlkYFk24PBdViImAG6UkxTDmeTitwEE6/RaxztRm9E0RKgDyi1e7xtLHGaggcalBw==","hashingAlgo":"sha256","signingAlgo":"rsa2048"}
[DEBG] 2024-07-31 15:41:33.189567 | sshnpd | Executing handle_ssh_request
[DEBG] 2024-07-31 15:41:33.189678 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:33.219744 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:33.219821 | connection | 	SENT: "plookup:bypassCache:true:all:publickey@soccer0"
[DEBG] 2024-07-31 15:41:33.307144 | connection | 	RECV: "data:{"data":"MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAoIFOnmCE1b4bZCBfbajiwg3GXqW8syOJhx/5bhvxKA8I21U1IlKVw8CQ3mLSNQl4+R1cWUrhjvw1EYBtTQdQgr5EF72bXB8k+mCoXHqu7YzyD04ECdc6GFPZrzAH2TgvkYHTuOytXCye4tHr/4pZxah1NjNwH7ULt78JQIl/me6NjSTBRh6F/AdpXy1ZCR5vO8KuXY14JVhwWSZZvSli/gAIDyTmudimjvlHpI9wdxmht91WSZAsJpORZAZAm9WtCsAbyH8ggUbXlzhwNqtSnqbe5JeoK+I5fxn3ISRygF3uSncknmd2okcs5nILWb0CafJi8w7n7RsbMnmn3vjfCwIDAQAB","metaData":{"createdBy":"@soccer0","updatedBy":"@soccer0","createdAt":"2024-02-26 14:18:11.849Z","updatedAt":"2024-02-26 14:19:00.302Z","status":"active","version":2,"ttl":86400000,"isBinary":false,"isEncrypted":false},"key":"public:publickey@soccer0"}"
[DEBG] 2024-07-31 15:41:33.307773 | SSH_REQUEST | verify_envelope_signature (success)
[DEBG] 2024-07-31 15:41:33.330659 | RUN SRV | Starting srv with command: srv -h 85.239.63.180 -p 39923 --local-port 22 --local-host localhost --rv-auth --rv-e2ee
[INFO] 2024-07-31 15:41:33.331185 | srv - run | Configuring encrypter/decrypter for srv
[INFO] 2024-07-31 15:41:33.331432 | srv - run | Starting socket to socket srv
[INFO] 2024-07-31 15:41:33.331616 | srv - run | Initializing connection for side a
[INFO] 2024-07-31 15:41:33.331815 | srv - side | Doing tcp connect to localhost:22
[INFO] 2024-07-31 15:41:33.340570 | srv - run | Initializing connection for side b
[INFO] 2024-07-31 15:41:33.340708 | srv - side | Doing tcp connect to 85.239.63.180:39923
[DEBG] 2024-07-31 15:41:33.330504 | SSH_REQUEST | Appended the srv process id node to the end of the process queue[DEBG] 2024-07-31 15:41:33.351747 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:33.382464 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:33.382557 | connection | 	SENT: "llookup:shared_key.soccer0@soccer99"
[DEBG] 2024-07-31 15:41:33.413888 | connection | 	RECV: "data:GRluAnPbZML0MVUCoq/MWwv0ZVZyRNigrKoeMyn2pZxBujbfPk+KAoGmOmwvEJGsYuplAKZLLreV3QcQxHxUaHQ/weWwXMrgaogJkUlSt99fUAl/vZmK0hG31tnyJ4suEQTJHAUlmc09fy1mLlP/oBZuRLijL9aH0XirGsd7wAoFFuNNaKwhN0Lx8mVjrQNqq4uY39pJuMWAPebLG6uNQk6PZX8GQ9tu15YGerJqRJUsVrXVkPq3jNmRMaMjZWbluz/bzD7b7BK207WqiMv938awGD0qfixPjVg8gY2kY+8uMMXYlVe8eZz4C0LC/egFeEXJkidY2FeNCeJ8oIMtgQ=="
[INFO] 2024-07-31 15:41:33.415165 | srv - run | Starting threads
[INFO] 2024-07-31 15:41:33.415484 | srv - run | Sending auth string
rv started successfully
[DEBG] 2024-07-31 15:41:33.428231 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:33.458439 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:33.458683 | connection | 	SENT: "notify:update:messageType:key:priority:low:strategy:all:ttln:86400000:ttl:10000:isEncrypted:true:ivNonce:BNVbd/4ox4a4ahE67xDL5w==:@soccer0:28dee05d-ce71-4d24-af38-638233a495e4.jeremyvps.sshnp@soccer99:B+ePUUNdloY7CK8BhiGGRXrdDMgRId8nuGH0HXgN330ECBWjg3pgMw0c2cQ0Gy7fKgbN/J5PRAxtT8PAAifwqntKldJgu9GamEk5ttdOFnpDAiKIms9oLbSN7A55OkaK0iGVIeheivas4ip0wA4AYcrpQawjxoh1v5uzg9tyerXsEfPGzbXS36TbPx+/TDEWB1P0/pDMU3JLpRdNKLBiODsbSsrwyCvRJsXx3mpMNQL8AbWCABrAohosGYOREkm0g8qJg0MXSC80K10oR/h4GYQ2D6Gm2Y5ZOV4hV4x6suoX9Oza20ZF6QTctvSA+srCMVh/wp64g2GbBDlHLocaOQ27jGeeZyec815ENZcurGobXOWtmItSBBkK5PExzK95YaTFsPjJwYfRfmyx8BWMxOaqNoX5QSpF3rwTlzZmrxJo49/KJRb75OcvG9Goj5Jm3J/DA8fm/rTZpbUeLX6pevVqkDO/z8hQHoc65WykknBGqxUPZrtqkKqGJuMiuigWXg0FB4vFsUqVqXJ5hkkd8NueUwilXkzk0xtKWZEQ1HBoOgz2aOpwnlZgq1jltq746loJ11vdGhT7wJJO2s4skgL5I9aejBjxKrbXHbrErnJdHjWFP7s2YafL9WDxIjdybqRLuY7U3G5QqizMrRGPvVakTvUSUwxTCpqGmRch8URLXZV5Ty9gW3acVKT0TGu8+Aqls8lCu5vs35pszGN0tBGP/3SpGFcnfZnvguNef6rLEW1iW/DyWVHNTcIgAGryoH+LZrkdq72hwrKvLtYev0cmKQf8OxDZoMMIJbl6hr9jyjeDI4CWY6VtcpmiSCbSEgE9q9Nm20s7/u0ILjcP8M1Q8O2ferFUHLVuCVAdx+ou25kUoAIe2b+2tYYTQ+9ClgTReKXhjFFfzMFDY0vJfiKoxhgzAlrlNdTGqYjCj1zpQP89GBY8ps0sumFoAITAiHLYIhazkD8MRLgJCyCEZCrs/1AV36zY4V68CkHw1G6DQbZfAZPjiAONbbc5Yb9TIglqPrAz4EvBO62NTmtk+qtq1Iw7RR/rtofys+rmJWtylrb77X0htSNA/TQAqvpRDRRFKQcrE8lNkBnfbQvaEVQdgg3uC3xdLf40EdL3T91mEYsAQKI9r2zM9/LM6FoAUB1zn8KWCBQ1uyhb3bBe/zfpPPIrw3RI4na3ca3DmmTRbrns7gz4eRpi8/Y7WLmYvkVNb0Ij6dkp2/AcgERhH1RyhhOMCOBTULl2gcKTw88afkBWagwLoXxQTOds2bMztv1UR8qcc4m4SupMJHSxNpz1gIMTek9FJ7Bw/UpGEn2SniyT8L551eqkmnVUHKblvYeVJCiDXwRbiOfFOzUlj/UY78AXEQAQ8dbk3TlP81N+RLccoZ7to8AnaeK8fmIJEEDtXw72+OHgACuIXHl079IqPoeOWce5grMz8BFMsClAQgE5GehgyFgswJaF1utWgj2Ax+/4vYEPE8dmMcmajrmfO9HLOmMIqSQq/wr07DcYLAE+K4c0M649Ulxmw8QYCrGUOsbjN9jg9v8EbQlkoa6ZBZAs8l4F6jLX8MwLNdZ1ZR0XJ+BhEmbHb3tvb3Ty6dZ1qhLhOMk47SvUV8msUdN7O39MAKWHm7MTt/j9tEty2HgqtQzpw4a75pqddd9b"
[DEBG] 2024-07-31 15:41:33.489841 | connection | 	RECV: "data:a21d82b1-b9d4-49f7-a2c9-9f5c4f2c70e5"
[DEBG] 2024-07-31 15:41:33.489888 | SSH_REQUEST | Released the atclient lock
[DEBG] 2024-07-31 15:41:33.489963 | sshnpd | Waiting for next monitor thread message
[INFO] 2024-07-31 15:41:35.649518 | sshnpd | Exited main loop
[DEBG] 2024-07-31 15:41:36.493061 | atclient_monitor | Read nothing from the monitor connection: -26624
[DEBG] 2024-07-31 15:41:36.493112 | sshnpd | Received message of type: 4
[DEBG] 2024-07-31 15:41:36.493190 | connection | 	SENT: "noop:0"
[DEBG] 2024-07-31 15:41:36.523999 | connection | 	RECV: "data:ok"
[DEBG] 2024-07-31 15:41:36.524051 | sshnpd | Waiting for next monitor thread message
[DEBG] 2024-07-31 15:41:38.536920 | atclient_monitor | 	RECV: "{"id":"-1","from":"@soccer99","to":"@soccer99","key":"statsNotification.@soccer99","value":"914","operation":"update","epochMillis":1722440498466,"messageType":"MessageType.key","isEncrypted":false,"metadata":null}"
```

File: .github/workflows/c_release.yml [hunk 3]
```diff
@@ -29,14 +29,22 @@ jobs:
...
             ext: ""
             bundle: "shell"
+            compiler: "clang"
+            flags: "-Wno-error -pthread"
           - os: macos-14
             output-name: sshnp-macos-arm64
             ext: ""
             bundle: "shell"
+            compiler: "clang"
+            flags: "-Wno-error -pthread"
```
Comment (xavierchanth) [inline .github/workflows/c_release.yml:47]: CC: @cpswan

**Pull request merged**